### PR TITLE
Add explicit Python 3.13 support

### LINKED ISSUE #162: Certificate chain APIs may change in Python 3.13
Didn't test this directly, but I'm aware that the certificate chain APIs may change in 3.13. Might be a good idea to test to make sure [the current usage](https://github.com/elastic/elastic-transport-python/blob/main/elastic_transport/_node/_urllib3_chain_certs.py#L113) doesn't break.

### PR DESCRIPTION
**Pull request by pquentin**
Closes #162

### CODE REVIEW

File: elastic_transport/_node/_http_aiohttp.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ async def perform_request(  # type: ignore[override]
                 data=body_to_send,
                 headers=request_headers,
                 timeout=aiohttp_timeout,
-                **kwargs,
+                **kwargs,  # type: ignore[arg-type]
```
Comment (pquentin) [inline elastic_transport/_node/_http_aiohttp.py:185]: This is not related to Python 3.13, but caused instead by https://github.com/aio-libs/aiohttp/pull/8463. I.. just have no idea how to make mypy happy here, short of maybe adding a class?

File: elastic_transport/_node/_http_urllib3.py [hunk 2]
```diff
@@ -199,7 +199,7 @@ def perform_request(
                 body=body,
                 exception=err,
             )
-            raise err from None
+            raise err from e
```
Comment (pquentin) [inline elastic_transport/_node/_http_urllib3.py:202]: This is also unrelated, but it feels wrong to hide parts of exceptions. (And this was needed to debug chain certs code.)

**Review by miguelgrinberg — APPROVED** [review]
